### ISSUE TITLE
DB2 expects Double instead of Integer as parameter for pagination

### ISSUE THREAD

**Issue by DavideD** [eclipse-vertx/vertx-sql-client#1303]
In Db2, with Vert.x 4.3.8 and 4.4.1, I have the following table:
```
create table Flour (id integer not null, description varchar(255), name varchar(255), type varchar(255), primary key (id))
```

And I'm trying to run the following query for pagination:
```
select *
from (
    select f1_0.id c0,f1_0.description c1,f1_0.name c2,f1_0.type c3,row_number() over(order by f1_0.id) rn
    from Flour f1_0 ) r_0_
where r_0_.rn<=?+? and r_0_.rn>? order by r_0_.rn
```

with the following list of parameters (`ListTuple`):
```
0 = {Integer@9248} 1
1 = {Integer@9248} 1
2 = {Integer@9248} 1
```

And I get the following exception:
```vert
Caused by: io.vertx.core.impl.NoStackTraceThrowable: Parameter at position[0] with class = [java.lang.Integer] and value = [1] can not be coerced to the expected class = [java.lang.Double] for encoding.
```

Why is it expecting a double?

**Comment by DavideD** [comment]
I have a similar error for this query:
```
select * from (select f1_0.id c0,f1_0.description c1,f1_0.name c2,f1_0.status c3,row_number() over(order by f1_0.id) rn from FamousPerson f1_0 where f1_0.name = ?) r_0_ where r_0_.rn>? order by r_0_.rn
```

But now it expects a Long:
```
[2023-04-04 09:30:38] SEVERE io.vertx.db2client.impl.codec.DB2Encoder                FATAL: Unable to encode command: io.vertx.sqlclient.impl.command.ExtendedQueryCommand@7d971e2c
java.lang.ClassCastException: class java.lang.Integer cannot be cast to class java.lang.Long (java.lang.Integer and java.lang.Long are in module java.base of loader 'bootstrap')
	at io.vertx.db2client.impl.drda.DRDAQueryRequest.buildFDODTA(DRDAQueryRequest.java:790)
	at io.vertx.db2client.impl.drda.DRDAQueryRequest.buildSQLDTAcommandData(DRDAQueryRequest.java:668)
	at io.vertx.db2client.impl.drda.DRDAQueryRequest.writeOpenQuery(DRDAQueryRequest.java:335)
	at io.vertx.db2client.impl.codec.ExtendedQueryCommandBaseCodec.encodePreparedQuery(ExtendedQueryCommandBaseCodec.java:53)
	at io.vertx.db2client.impl.codec.ExtendedQueryCommandCodec.encodeQuery(ExtendedQueryCommandCodec.java:36)
	at io.vertx.db2client.impl.codec.QueryCommandBaseCodec.encode(QueryCommandBaseCodec.java:53)
	at io.vertx.db2client.impl.codec.DB2Encoder.write(DB2Encoder.java:76)
	at io.vertx.db2client.impl.codec.DB2Encoder.write(DB2Encoder.java:60)
	at io.netty.channel.CombinedChannelDuplexHandler.write(CombinedChannelDuplexHandler.java:346)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:879)
	at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968)
	at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856)
	at io.vertx.sqlclient.impl.SocketConnectionBase.lambda$prepareCommand$4(SocketConnectionBase.java:269)
	at io.vertx.sqlclient.impl.command.CommandResponse.fire(CommandResponse.java:46)
	at io.vertx.sqlclient.impl.SocketConnectionBase.handleMessage(SocketConnectionBase.java:292)
	at io.vertx.sqlclient.impl.SocketConnectionBase.lambda$init$0(SocketConnectionBase.java:105)
	at io.vertx.core.impl.EventLoopContext.emit(EventLoopContext.java:55)
	at io.vertx.core.impl.ContextBase.emit(ContextBase.java:239)
	at io.vertx.core.net.impl.NetSocketImpl.handleMessage(NetSocketImpl.java:378)
	at io.vertx.core.net.impl.ConnectionBase.read(ConnectionBase.java:158)
	at io.vertx.core.net.impl.VertxHandler.channelRead(VertxHandler.java:153)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.vertx.db2client.impl.codec.DB2Encoder.lambda$write$0(DB2Encoder.java:72)
	at io.vertx.db2client.impl.codec.PrepareStatementCodec.handleReadyForQuery(PrepareStatementCodec.java:91)
	at io.vertx.db2client.impl.codec.PrepareStatementCodec.handleColumnDefinitionsDecodingCompleted(PrepareStatementCodec.java:103)
	at io.vertx.db2client.impl.codec.PrepareStatementCodec.decodePayload(PrepareStatementCodec.java:82)
	at io.vertx.db2client.impl.codec.DB2Decoder.decodePayload(DB2Decoder.java:79)
	at io.vertx.db2client.impl.codec.DB2Decoder.decode(DB2Decoder.java:52)
	at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by vietj** [comment]
it seems to be in the DRDA layer, so maybe a bug there

**Comment by tsegismont** [comment]
Fixed in 74c74cd9

**Comment by DavideD** [comment]
Awesome, thanks!
